src/depot_map.h

```diff
@@ -43,16 +43,21 @@ static inline bool IsDepotTile(Tile tile)
 	return IsRailDepotTile(tile) || IsRoadDepotTile(tile) || IsShipDepotTile(tile) || IsHangarTile(tile);
 }
 
+extern DepotID GetHangarIndex(TileIndex t);
+
 /**
  * Get the index of which depot is attached to the tile.
  * @param t the tile
- * @pre IsRailDepotTile(t) || IsRoadDepotTile(t) || IsShipDepotTile(t)
+ * @pre IsDepotTile(t)
  * @return DepotID
  */
 static inline DepotID GetDepotIndex(Tile t)
 {
-	/* Hangars don't have a Depot class, thus store no DepotID. */
-	assert(IsRailDepotTile(t) || IsRoadDepotTile(t) || IsShipDepotTile(t));
+	assert(IsDepotTile(t));
+
+	/* Hangars don't store depot id on m2. */
+	if (IsTileType(t, MP_STATION)) return GetHangarIndex(t);
```

Comment:
The depot id for rail, road and ship depots is stored in m2, as the station id.
As airport hangars are part of the airport, they should store the station id and the depot id.
When I wrote this, there was no available space for storing it in the tile. Storing the depot_id of hangars in m8 is now possible, but then it is not fully uniform, as some depot_ids will be stored in m2 and some others in m8.

I also thought about storing it in m8 some time ago, but as it is now (using GetHangarIndex) seems ok to me. It may look ugly, but having it under Station::Airport::DepotID is useful enough. Even if the depot_id was also stored in the map array, it is usually better to have it under Airport::DepotID, as it is something aircraft will need while they probably have a pointer to the corresponding station. Moreover, while toying with the idea of modular airports, the more free bits on the map array, the better.

Reply:
I am not sure whether to store the depot index in the hangar tile. I agree that it would make things sightly more uniform, but I would prefer not to use those bits.
Any suggestion/preference about which approach should I take?

Reply:
Your reasoning makes sense to me, unless @rubidium42 disagrees.